### PR TITLE
Port foreman_subnet to apypie

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: plugins/modules/foreman_subnet.py [hunk 1]
```diff
@@ -221,37 +208,34 @@ def main():
...
 
-    for feature in ('dhcp_proxy', 'tftp_proxy', 'discovery_proxy', 'dns_proxy', 'remote_execution_proxies'):
-        if feature in entity_dict:
-            entity_dict[feature] = find_smart_proxy(module, entity_dict[feature])
+        if 'domains' in entity_dict:
+            entity_dict['domains'] = module.find_resources('domains', entity_dict['domains'], thin=True)
 
-    if 'organizations' in entity_dict:
-        entity_dict['organizations'] = find_organizations(module, entity_dict['organizations'])
+        # TODO should remote_execution seach for a list?
```
Comment (mdellweg) [inline plugins/modules/foreman_subnet.py]: @bagasse What do you think? One Problem is, it is not even in the foreman.json atm.

Comment (evgeni) [reply plugins/modules/foreman_subnet.py]: I think it should, and the UI allows that. But how to do that I don't know :)

File: tests/test_playbooks/vars/subnet.yml [hunk 2]
```diff
@@ -4,7 +4,6 @@ subnet_orgs:
 - Test Org2
 subnet_locs:
 - Foo
-- Foo/Baz
```
Comment (evgeni) [inline tests/test_playbooks/vars/subnet.yml]: Why?

Comment (mdellweg) [reply tests/test_playbooks/vars/subnet.yml]: I guess, we look up the locations in the wrong way. I guess, i should fix that.

**Comment by evgeni** [comment]
Did you by any chance have seen, reproduced and fixed https://github.com/theforeman/foreman-ansible-modules/issues/305 in this PR? :)

**Comment by mdellweg** [comment]
> Did you by any chance have seen, reproduced and fixed #305 in this PR? :)

I will have a try.

**Comment by evgeni** [comment]
Actually, I think I have a fix.

File: tests/test_playbooks/vars/subnet.yml (hunk 2 above)
Comment (mdellweg) [reply tests/test_playbooks/vars/subnet.yml]: If we want to search locations by title, i urge you to ack/merge #332 .

File: plugins/modules/foreman_subnet.py [hunk 3]
```diff
@@ -220,37 +207,34 @@ def main():
...
-        # Try to find the Subnet to work on
-        entity = find_subnet(module, name=entity_dict['name'], failsafe=True)
-    except Exception as e:
-        module.fail_json(msg='Failed to find entity: %s ' % e)
-
-    if 'mask' in entity_dict and 'cidr' not in entity_dict:
-        entity_dict['cidr'] = IPNetwork('%s/%s' % (entity_dict['network'], entity_dict['mask'])).prefixlen
-    elif 'mask' not in entity_dict and 'cidr' in entity_dict:
-        entity_dict['mask'] = IPNetwork('%s/%s' % (entity_dict['network'], entity_dict['cidr'])).netmask
+    # Try to find the Subnet to work on
```
Comment (evgeni) [inline plugins/modules/foreman_subnet.py]: I wonder if this comment is really worth it now, especially as the try/except is hidden from this code.

Comment (mdellweg) [reply plugins/modules/foreman_subnet.py]: :scissors:

**Comment by evgeni** [comment]
travis is green: https://travis-ci.org/theforeman/foreman-ansible-modules/builds/557828295

**Pull request merged**